catchup/service.go

```diff
@@ -345,6 +342,30 @@ func (s *Service) pipelinedFetch(seedLookback uint64) {
 		close(completed)
 	}()
 
+	var peerSelector *peerSelector
+	if s.cfg.NetAddress != "" { // Relay node
+		peerSelector = makePeerSelector(
+			s.net,
+			[]peerClass{
+				{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersConnectedOut},
+				{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookArchivers},
+				{initialRank: peerRankInitialThirdPriority, peerClass: network.PeersPhonebookRelays},
+				{initialRank: peerRankInitialFourthPriority, peerClass: network.PeersConnectedIn},
+			})
+	} else {
+		peerSelector = makePeerSelector(
+			s.net,
+			[]peerClass{
+				{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookArchivers},
+				{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersConnectedOut},
+				{initialRank: peerRankInitialThirdPriority, peerClass: network.PeersPhonebookRelays},
+			})
+	}
```

Comment:
nit: refactor out these `makePeerSelector` call into external method `createPeerSelector`, that would get an argument `pipelineFetch bool`. Internally, the method would determine the best set of peer classes, and would call `makePeerSelector`, returning the peerSelector object.
i.e.
```
func (s *Service) makePeerSelector(createPeerSelector bool) *peerSelector {
   var peerClasses []peerClass
   if createPeerSelector {
      if s.cfg.NetAddress != "" { // Relay node
          peerClasses = ...
      } else {
      }
   } else {
      ...
   }
   return makePeerSelector(s.net, peerClasses)
}
```

Reply:
done